authentication/views/profile.py

```diff
@@ -1,50 +1,96 @@
+
+    SIMPLE_USERS_EDITABLE_FIELDS = ['nickname',
+                                    'phone',
+                                    'room',
+                                    'address',
+                                    'city_of_origin',
+                                    'option',
+                                    'sports',
+                                    'roommate',
+                                    'minesparent', ]
+    """The profile fields editable by a simple (non-admin) user."""
 
-class UserViewSet(viewsets.ModelViewSet):
-    """
-    API endpoint that allows user to be viewed or edited.
-    """
     queryset = User.objects.all()
     serializer_class = UserSerializer
+    permission_classes = [ProfilePermission]
 
     filter_backends = (SearchFilter, DjangoFilterBackend)
     search_fields = ('id', 'first_name', 'last_name')
     filter_fields = ('promo',)
+    pagination_class = ProfileViewSetPagination
 
     def get_object(self):
         """
-        This method is called by the view set to retrieve
-        data for a given user. The user identifier is in the pk kwargs.
-
-        If pk is the string "current", we return the data for the current
-        authenticated user.
+        This method is called by the view set to retrieve data for a given user.
+        The user identifier is in the pk kwargs.\n
+        If pk is the string "current", return the data for the current authenticated user.
         """
-        pk = self.kwargs.get('pk')
-        if pk == "current":
-            return self.request.user
 
-        return super(UserViewSet, self).get_object()
+        if self.kwargs['pk'] == 'current':
+            self.check_object_permissions(self.request, self.request.user)
+            return self.request.user
 
-    def get_queryset(self):
-        queryset = User.objects.all()
-        starts_with = self.request.query_params.get('startswith', None)
-        quantity = int(self.request.query_params.get('quantity', 10))
-        if quantity > 10:
-            raise ValidationError("You can't request for more than 10 elements")
-        if starts_with is not None:
-            queryset = queryset.filter(id__startswith=starts_with) | queryset.filter(first_name__startswith=starts_with) | queryset.filter(last_name__startswith=starts_with)
-            queryset = queryset[:quantity]
-        return queryset
+        return super(ProfileViewSet, self).get_object()
 
     def get_serializer_class(self):
-        if self.action == 'list' and self.request.query_params.get('startswith') is not None:
+        if self.action in ('update', 'partial_update'):
+            return UpdateUserSerializer
+        elif self.action in ('list', ):
             return UserShortSerializer
         else:
             return UserSerializer
+
+    def update(self, request, *args, **kwargs):
+        """
+            Can X update Y? \n
+            X / Y       | Herself                               | Another user | \n
```

Comment:
Only 👩 ?

Reply:
I always thought you had to use feminine in English when you didn't know the gender :D

Reply:
I always thought you had to use feminine in English when you didn't know the gender :D

Reply:
Nope, use `them` ;-)

Reply:
Looks like it's better to use _they_/_their/…
https://english.stackexchange.com/questions/48/is-there-a-correct-gender-neutral-singular-pronoun-his-vs-her-vs-their